app/src/main/java/at/roteskreuz/stopcorona/model/db/dao/SessionDao.kt

```diff
@@ -19,30 +16,37 @@ abstract class SessionDao {
     protected abstract suspend fun insertSession(session: DbSession): Long
 
     @Insert
-    protected abstract suspend fun insertFullBatch(batch: DbFullBatchPart): Long
+    protected abstract suspend fun insertFullBatchPath(batch: DbFullBatchPart): Long
 
     @Insert
-    protected abstract suspend fun insertDailyBatch(batch: DbDailyBatchPart): Long
-
-    @Query("DELETE FROM session WHERE currentToken = :token")
-    abstract suspend fun deleteSession(token: String): Int
+    protected abstract suspend fun insertDailyBatchPath(batch: DbDailyBatchPart): Long
 
     @Transaction
-    open suspend fun insertOrUpdateFullSession(fullSession: DbFullSession) {
-        val token = fullSession.session.currentToken
-        deleteSession(token)
-        insertSession(fullSession.session)
-        fullSession.fullBatchParts.forEach { fullBatchPath ->
-            fullBatchPath.id = 0
-            insertFullBatch(fullBatchPath.copy(currentToken = token))
+    open suspend fun insertFullSession(fullSession: DbFullSession) {
+        val session = fullSession.session
+        deleteSession(session)
```

Comment:
Old session won't be deleted since id is autogenerated and `fullSession` is a new instance. What do you want to achieve here?

Reply:
we just want to get a new, clean session.
sessions are deleted when they are done processing. This should even be implemented exception safe.

Reply:
You are right. This is a left over. the method used to be `insertOrUpdateFullSession(token: String)`
Will remove

Reply:
done